internal_tools/hw_decoder_bench.py

```diff
@@ -90,368 +129,406 @@
+                    + f"Throughput: {throughput:.2f} frames/sec "
+                    + f"(mean={mean_t:.6f}sec, median={median_t:.6f}sec, "
+                    + f"min={min_t:.6f}sec, max={max_t:.6f}sec)"
+                )
+
+        end_time = time.perf_counter()
+        total_time = end_time - start_time
+        total_throughput = test_iterations * args.batch_size * args.gpu_num / total_time
+        avg_t = statistics.mean(execution_times)
+        stdev_t = statistics.stdev(execution_times)
         median_t = statistics.median(execution_times)
         min_t = min(execution_times)
         max_t = max(execution_times)
-        print(
-            f"Iteration {iteration + 1}/{test_iterations} - "
-            + f"Throughput: {throughput:.2f} frames/sec "
-            + f"(mean={mean_t:.6f}sec, median={median_t:.6f}sec, "
-            + f"min={min_t:.6f}sec, max={max_t:.6f}sec)"
+
+        print("\nFinal Results:")
+        print(f"CPU threads: {cpu_num}")
+        print(f"HW decoder load: {hw_load}")
+        print(f"Total Execution Time: {total_time:.6f} sec")
+        print(f"Total Throughput: {total_throughput:.2f} frames/sec")
+        print(f"Average time per iteration: {avg_t:.6f} sec")
+        print(f"Median time per iteration: {median_t:.6f} sec")
+        print(f"Stddev time per iteration: {stdev_t:.6f} sec")
+        print(f"Min time per iteration: {min_t:.6f} sec")
+        print(f"Max time per iteration: {max_t:.6f} sec")
+        perf_results.append(
+            {
+                "cpu_num": cpu_num,
+                "hw_load": hw_load,
+                "total_time": total_time,
+                "total_throughput": total_throughput,
+            }
         )
 
-end_time = time.perf_counter()
-total_time = end_time - start_time
-total_throughput = test_iterations * args.batch_size * args.gpu_num / total_time
-avg_t = statistics.mean(execution_times)
-stdev_t = statistics.stdev(execution_times)
-median_t = statistics.median(execution_times)
-min_t = min(execution_times)
-max_t = max(execution_times)
-
-print("\nFinal Results:")
-print(f"Total Execution Time: {total_time:.6f} sec")
-print(f"Total Throughput: {total_throughput:.2f} frames/sec")
-print(f"Average time per iteration: {avg_t:.6f} sec")
-print(f"Median time per iteration: {median_t:.6f} sec")
-print(f"Stddev time per iteration: {stdev_t:.6f} sec")
-print(f"Min time per iteration: {min_t:.6f} sec")
-print(f"Max time per iteration: {max_t:.6f} sec")
+if len(perf_results) > 0:
+    # Find and print result with best throughputif
+    best_result = max(perf_results, key=lambda x: x["total_throughput"])
+    print("\nBest throughput configuration:")
+    print(f"CPU threads: {best_result['cpu_num']}")
+    print(f"HW decoder load: {best_result['hw_load']}")
+    print(f"Total time: {best_result['total_time']:.6f} sec")
+    print(f"Throughput: {best_result['total_throughput']:.2f} frames/sec")
+else:
+    print("No results to display")
```

Comment:
an idea, print the results for all parameters in a way that can be easily plotted.

Reply:
Can you elaborate which format you have in mind?

Reply:
It was just a thought. Instead of just printing the best result, print all the values for all the tested parameters, so that we can plot them. Anyway, feel free to consider this as out-of-scope for now

Reply:
We print them as the benchmark goes, do you think it is better to print them again at the end?